papis/commands/doctor.py

```diff
@@ -0,0 +1,169 @@
+import papis.strings
+import papis.document
+
+
+Error = NamedTuple("Error", [("name", str),
+                             ("path", str),
+                             ("msg", str),
+                             ])
+TestFn = Callable[[papis.document.Document], List[Error]]
+Test = NamedTuple("Test", [("name", str),
+                           ("operate", TestFn),
+                           ("suggest_cmd", Callable[[Error], str])])
+
+logger = logging.getLogger("doctor")
+
+
+def register_test(name: str, test: Test) -> None:
+    """
+    Register a test.
+    To be used by users in their configuration files
+    for example.
+    """
+    REGISTERED_TESTS[name] = test
+
+
+def files_test(doc: papis.document.Document) -> List[Error]:
+    """
+    It checks wether the files of a document actually exist in the
+    filesystem.
+    """
+    files = doc.get_files()
+    results = []  # type: List[Error]
+    folder = doc.get_main_folder()
+    for _f in files:
+        if not os.path.exists(_f):
+            results.append(Error(name="files",
+                                 path=folder or "",
+                                 msg=_f))
+    return results
+
+
+def keys_test(doc: papis.document.Document) -> List[Error]:
+    """
+    It checks wether the keys provided in the configuration
+    option ``doctor-keys-test`` exit in the document.
+    """
+    keys = papis.config.getlist("doctor-keys-test")
+    folder = doc.get_main_folder()
+    results = []  # type: List[Error]
+    for k in keys:
+        if k not in doc or len(doc[k]) == 0:
+            results.append(Error(name="keys",
+                                 path=folder or "",
+                                 msg=k))
+    return results
+
+
+REGISTERED_TESTS = {
+    "files": Test(operate=files_test,
+                  name="test",
+                  suggest_cmd=lambda e:
+                  """
+                  papis edit --doc-folder {}
+                  """.format(e.path)),
+    "keys": Test(operate=keys_test,
+                 name="keys",
+                 suggest_cmd=lambda e:
+                 """
+                 papis update --doc-folder {}
+                 """.format(e.path)),
+}  # type: Dict[str, Test]
+
+
+def run(doc: papis.document.Document, tests: List[str]) -> List[Error]:
+    """
+    Runner for doctor. It runs all the tests given by the test
+    argument, and it gets the test from the global REGISTERED_TESTS
+    dictionary.
+    """
+    results = []  # type: List[Error]
+    for test in tests:
+        results += REGISTERED_TESTS[test].operate(doc)
+    return results
+
+
+@click.command("doctor")
+@click.help_option("--help", "-h")
+@papis.cli.query_option()
+@papis.cli.sort_option()
+@click.option("-t", "--tests", "_tests",
+              default=lambda: papis.config.getlist("doctor-default-tests"),
+              multiple=True,
+              help=("Tests to run on every document, possible values: {}"
+                    .format(REGISTERED_TESTS)))
```

Comment:
```suggestion
                    .format(", ".join(t.name for t in REGISTERED_TESTS)))
```
? The helps looks very heavy with the whole object.